crypto/mem_test.cc

```diff
@@ -20,7 +20,12 @@ extern "C" {
   OPENSSL_EXPORT void OPENSSL_memory_free(void *ptr);
   OPENSSL_EXPORT size_t OPENSSL_memory_get_size(void *ptr);
   OPENSSL_EXPORT void *OPENSSL_memory_realloc(void *ptr, size_t size);
+
+  OPENSSL_EXPORT void *new_malloc_impl(size_t size, const char *file, int line);
+  OPENSSL_EXPORT void new_free_impl(void *ptr, const char *file, int line);
+  OPENSSL_EXPORT void *new_realloc_impl(void *ptr, size_t size, const char *file, int line);
```

Comment:
Here too, I think these can just be regular functions down below and then pas them in for the test.

Reply:
It seems needed. 
```
../crypto/mem_test.cc: In function ‘void* new_malloc_impl(size_t, const char*, int)’:
../crypto/mem_test.cc:100:7: error: no previous declaration for ‘void* new_malloc_impl(size_t, const char*, int)’ [-Werror=missing-declarations]
 void *new_malloc_impl(size_t size, const char *file, int line) {
       ^~~~~~~~~~~~~~~
../crypto/mem_test.cc: In function ‘void* new_realloc_impl(void*, size_t, const char*, int)’:
../crypto/mem_test.cc:104:7: error: no previous declaration for ‘void* new_realloc_impl(void*, size_t, const char*, int)’ [-Werror=missing-declarations]
 void *new_realloc_impl(void *ptr, size_t size, const char *file, int line) {
       ^~~~~~~~~~~~~~~~
```